### DISCUSSION TITLE
How to build standalone worker

### DISCUSSION THREAD

**Discussion by zdm** [webpack/webpack#14349] [Q&A] [answered]
Hi,
I am unable to find clear instructions, how to build completely separated js file to use as worker.
I tried to add additional entry point:
`webpack.config.js`:
```
...
entry: {
    "app": "./src/main.js",
    "firebase": {
        "import": "./src/firebase-messaging-sw.js",
         "filename": "firebase-messaging-sw.js",
          "publicPath": "/",
    },
},
...
```
This creates "/firebase-messaging-sw.js" file, but it depends on other chunks and can not be loaded as worker.

How I can bundle `firebase-messaging-sw.js` and all it's dependencies to the completely standalone js file?

**Comment by alexander-akait** [comment, marked as answer]
You can use multi compiler mode with `target: 'webworker'` and use https://webpack.js.org/configuration/other-options/#dependencies to make order of builds

**Comment by zdm** [comment]
Yes, thank you.
I already found this way.
